src/v/cluster/metadata_cache.cc

```diff
@@ -248,4 +276,47 @@ metadata_cache::get_default_shadow_indexing_mode() const {
     }
     return m;
 }
+
+ss::future<> metadata_cache::broadcast_disk_space_state(
+  const storage::disk_space_alert state) {
+    return container().invoke_on_others(
+      [state](metadata_cache& mc) { mc.set_disk_space_state(state); });
+}
+
+void metadata_cache::set_disk_space_state(const storage::disk_space_alert s) {
+    _disk_space_state = s;
+    vlog(clusterlog.debug, "set_disk_space_state {}", s);
+}
+
+bool metadata_cache::should_reject_writes() const {
+    return _disk_space_state == storage::disk_space_alert::degraded;
+}
+
+void metadata_cache::maybe_update_cluster_health() {
+    if (_refresh_gate.is_closed()) {
+        return;
+    }
+    ssx::spawn_with_gate(_refresh_gate, [this]() -> ss::future<> {
```

Comment:
I'm not sure that we need to manage the disk space information in the metadata cache. Ostensibly the metadata cache is for more kafka metadata, and we already have a health monitor, so perhaps we can expand the usage scenarios for the health monitor rather than the metadata cache?

I think there are two changes that would make this cleaner:

1. The health monitor should keep itself up-to-date, rather than relying on other modules to implement polling to trigger cluster health refresh (like what is done here with the timer). This can even be done in a hot path where the if a refresh is needed it is triggered and debounced inside the health monitor.
2. Instead of using an upcall from the health monitor to populate the metadata cache with disk space usage, can we query the health monitor directly? The information seems to already be there and the health monitor is generally available at the kakfa layer.

Reply:
The callback has been eliminated. I pushed an update implementing @mmaslankaprv 's suggestion. Can you resolve this thread and continue the discussion on the newer patch please?

Reply:
Next update eliminates logic in metadata_cache, but retains a trivial wrapper around health_monitor_frontend for the `should_reject_write()` query. I'm not sure how to access the health monitor directly from kafka layer (i.e. k/server/handlers/produce.cc), and didn't see any examples of this from a quick grep.